lib/utils/KalturaCouchbaseConnector.js

```diff
@@ -0,0 +1,379 @@
+let util = require('util');
+let Promise = require("bluebird");
+
+let base = require('../KalturaBase');
+
+let mock = false;
+let couchbase = require('couchbase');
+if (mock)
+{
+    couchbase = couchbase.Mock;
+}
+
+class KalturaCouchbaseConnector {
+
+    constructor(){
+        this._config = KalturaConfig.config.couchbase;
+        this._clusterName = this._config.clusterName;
+        this._bucketName = this._config.bucketName;
+        this._cluster = new couchbase.Cluster(this._clusterName);
+        this._bucket = this._cluster.openBucket(this._bucketName);
+        this._dataVersion = 0;
+
+        if (this._config.dataVersion)
+            this._dataVersion = parseInt(this._config.dataVersion);
+
+        process.on('unhandledRejection', (reason) => {
+            KalturaLogger.log(reason);
+        });
+    }
+
+    static getInstance()
+    {
+        if (typeof KalturaCouchbaseConnector.instance == 'undefined')
+        {
+            KalturaCouchbaseConnector.instance = new KalturaCouchbaseConnector();
+        }
+        return KalturaCouchbaseConnector.instance;
+    }
+
+    getDataVersion()
+    {
+        return this._dataVersion;
+    }
+
+    getStack()

```

Comment:
This function needs to be implemented once , in an util class. 
Currently it is implemented both in KalturaCouchbaseConnector  and KalturaCacheManager.
You can add it to a third class - KalturaErrorHandlerUtils or even to logger as a static function.